crates/evm/evm.rs

```diff
@@ -43,6 +47,60 @@ impl EvmState {
     }
 }
 
+//TODO:validate_block and execute_block should return declarative results and errors indicating the
+//     outcome of executing these functions.
+
+/// Performs pre-execution validation of the block's header values in reference to the parent_header
+/// Verifies that blob gas fields in the header are correct in reference to the block's body.
+/// If a block passes this check, execution will still fail with execute_block when a transaction runs out of gas
+pub fn validate_block(block: &Block, parent_header: &BlockHeader, state: &EvmState) -> bool {
+    //TODO: Fail if block is already on the blockchain
+    //TODO: Fail if the parent block identified by parent_hash is not present on the blockchain
+
+    let spec = spec_id(state.database(), block.header.timestamp).unwrap();
+
+    // Verify initial header validity against parent
+    let valid_header = match spec {
+        SpecId::CANCUN => {
+            validate_block_header(&block.header, parent_header)
+                && validate_cancun_header_fields(&block.header, parent_header)
+        }
+        _ => {
+            validate_block_header(&block.header, parent_header)
+                && validate_no_cancun_header_fields(&block.header)
+        }
+    };
```

Comment:
We could call `validate_block_header` outside of the match to avoid the repeated line

Reply:
done!